File: yh_turtle/scripts/turtle_keyboard.py
#!/usr/bin/python3
#-*- coding:utf-8 -*-
import rospy
from geometry_msgs.msg import Twist

class TurtleKeyboard:
    def __init__(self):
        super(TurtleKeyboard, self).__init__()
        # 서브스크라이버 선언
        self.sub = rospy.Subscriber("cmd_vel", Twist, self.msgCallback)
        # 퍼블리셔 선언
        self.pub = rospy.Publisher("turtle1/cmd_vel", Twist, queue_size=100)

    def msgCallback(self, msg):
        self.pub.publish(msg)

if __name__ == "__main__":
    rospy.init_node("turtle_keyboard")
    turtle_keyboard = TurtleKeyboard()
    rospy.spin() # 서브스크라이버가 메시지 받는걸 대기

File: yh_turtle/src/turtle_keyboard.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"

ros::Publisher pub;

void msgCallback(const geometry_msgs::Twist::ConstPtr& msg){
    pub.publish(*msg); //* : 포인터를 참조한다
}

int main(int argc, char** argv){
    
    ros::init(argc, argv, "turtle_keyboard");
    ros::NodeHandle nh;

    // 퍼블리셔 선언
    pub = nh.advertise<geometry_msgs::Twist>("turtle1/cmd_vel", 100);

    // 서브스크라이버 선언
    ros::Subscriber sub = nh.subscribe("cmd_vel", 100, msgCallback);

    ros::spin();
    
    return 0;
}